python/src/array.cpp

```diff
@@ -247,6 +247,15 @@ void init_array(nb::module_& m) {
           Returns:
             tuple(int): A tuple containing the sizes of each dimension.
         )pbdoc")
+      .def_prop_ro(
```

Comment:
That's probably not what you had in mind. Since everything in mlx is lazy there will be no strides available before the array is computed so this will be wrong unfortunately.

We 've had the discussion about including strides many times I think we should keep them unavailable from the public API. They are only useful inside an op implementation.

I would instead allow access to `<name of array argument>_[shape|strides|ndim]` any of which would be automatically included in the arguments if present in the kernel source.